sympy/ntheory/residue_ntheory.py

```diff
@@ -574,47 +574,39 @@ def is_nthpow_residue(a, n, m):
         return True
     if n == 2:
         return is_quad_residue(a, m)
-    return _is_nthpow_residue_bign(a, n, m)
-
-
-def _is_nthpow_residue_bign(a, n, m):
-    r"""Returns True if `x^n = a \pmod{n}` has solutions for `n > 2`."""
-    # assert n > 2
-    # assert a > 0 and m > 0
-    if primitive_root(m) is None or igcd(a, m) != 1:
```

Comment:
This condition is no longer used -- I don't understand the intricacies enough to understand why. Does this change the beahvior significantly in some way?

Reply:
If m is `2`, `4`, `q**k`, `2*q**k` (where `q` is an odd prime), then the primitive root exists. Therefore, in the call to `_is_nthpow_residue_bign(a, n, pow(p, k))` at line 604, the if statement at line 584 is always false. Given the above, the calculation can be done without calling `_is_nthpow_residue_bign` at line 604.

The necessary and sufficient condition for the root to exist is `all(_is_nthpow_residue_bign_prime_power(a, n, p, e) for p, e in factorint(m).items())`. The results are not different, just that they can be computed without calling `_is_nthpow_residue_bign_prime_power`, especially if the primitive roots are present.